packages/assets-controllers/src/AssetsContractController.test.ts

```diff
@@ -211,9 +242,10 @@ describe('AssetsContractController', () => {
 
   it('should throw missing provider error when getting ERC-20 token balance when missing provider', async () => {
     const { assetsContract, messenger } = await setupAssetContractControllers();
-    assetsContract.configure({ provider: undefined });
+    assetsContract.setProvider(undefined);
     await expect(
-      assetsContract.getERC20BalanceOf(
+      messenger.call(
```

Comment:
I see that you've changed all of the instances where we are calling a method on the class to call the corresponding messenger action. That makes sense — but it does create some inconsistency with the guidance we have for writing tests. The `describe`s for these tests use the name of the method, but here we are calling the messenger action. I worry that could be a bit confusing when reading these tests.

Do we want to change the `describe`s, or how do you think is the best way to resolve this discrepancy?

Reply:
Ah, you're right. We don't follow that `describe` convention here anyway.

Then I guess I will ask a slightly different question. Fast-forwarding, if we were to automatically register messenger actions for all public methods in all controllers, how should we go about writing tests? Would we ensure that the tests that currently center around method now center around the messenger actions (i.e. instead of calling the method we now call the messenger action in the test)? Would we test both the method and the messenger action by wrapping existing tests for a public method in a loop so that we can easily run the same tests for that method and the messenger action without having to copy those tests?

Reply:
I wonder if we could omit method tests if a messenger action handler has a single corresponding class method because it is defined using `this[method].bind(this)`? 

If we're only testing one of the two, it should be safer to test the messenger action, since the constraints for setting up a messenger action call (e.g. instantiating messenger and enumerating available actions) is a superset of the same for invoking its corresponding class method. 

Otherwise, the messenger action and the method should be equivalent, but to ensure that they don't become inconsistent due to code drift, we could add an `expect` call to these tests verifying that the action handler is a bound class method with no additional logic.

For other cases where a method has no corresponding messenger action (e.g. `NftController.watchNft`), or a messenger action handler is not directly derived from a class method (e.g. `ApprovalController:addRequest`), we will need to test, respectively, only the method, or both the method and the messenger separately.

What are your thoughts on testing only the messenger actions where the action and the method are equivalent? I feel like I might be missing some of the implications here e.g. how would this affect coverage?

Reply:
Nice that's our exact use case and it looks like it's a simple solution. We should definitely try applying this.